fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/village/v1/VillagerProfessionBuilder.java

```diff
@@ -0,0 +1,111 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.village.v1;
+
+import java.util.Set;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Item;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.village.TradeOffers;
+import net.minecraft.village.VillagerProfession;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import net.fabricmc.fabric.impl.object.builder.VillagerProfessionBuilderImpl;
+
+/**
+ * Allows the creation of new {@link VillagerProfession}s.
+ *
+ * <p>Note this does not register any trades to these villagers. To register trades, add a new entry with your profession as the key to {@link TradeOffers#PROFESSION_TO_LEVELED_TRADE}.
+ */
+public interface VillagerProfessionBuilder {
+	/**
+	 * Creates a builder to make a {@link VillagerProfession}.
+	 * @return A new builder.
+	 */
+	static VillagerProfessionBuilder builder() {
+		return new VillagerProfessionBuilderImpl();
+	}
+
+	VillagerProfessionBuilder id(Identifier id);
+
+	/**
+	 * The {@link PointOfInterestType} the villager will search for when finding it's workstation.
+	 * @param type The {@link PointOfInterestType}.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder workstationPointOfInterest(PointOfInterestType type);
+
+	/**
+	 * An item that a Villager may harvest in this profession.
+	 *
+	 * <p>In Vanilla this is used by the farmer to define what type of crops the farmer can harvest.
+	 *
+	 * @param item An item.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder gatherableItem(Item item);
+
+	/**
+	 * Set of items that a Villager may harvest in this profession.
+	 *
+	 * <p>In Vanilla this is used by the farmer to define what type of crops the farmer can harvest.
+	 *
+	 * @param items A set of items.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder gatherableItems(Set<Item> items);
+
+	/**
+	 * A block which might suffice as a job site for a Villager.
+	 *
+	 * <p>In vanilla this is used by the {@link VillagerProfession#FARMER Farmer} to stay near {@link Blocks#FARMLAND Farmland} when at it's job site.
+	 *
+	 * @param block The block.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder secondaryJobSite(Block block);
+
+	/**
+	 * Blocks which might suffice as a job site for a Villager.
+	 *
+	 * <p>In vanilla this is used by the {@link VillagerProfession#FARMER Farmer} to stay near {@link Blocks#FARMLAND Farmland} when at it's job site.
+	 *
+	 * @param blocks A set of blocks.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder secondaryJobSites(Set<Block> blocks);
+
+	/**
+	 * Represents the sound made when a Villager works.
+	 * @param workSoundEvent The {@link SoundEvent} to be played.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder workSoundEvent(SoundEvent workSoundEvent);
```

Comment:
`workSound`? Sound events for block settings are just done under the name `sounds`.